src/store/tree-picker/tree-picker-reducer.ts

```diff
@@ -2,25 +2,35 @@
 //
 // SPDX-License-Identifier: AGPL-3.0
 
-import { createTree, setNodeValueWith, TreeNode, setNode, mapTreeValues } from "~/models/tree";
+import { createTree, setNodeValueWith, TreeNode, setNode, mapTreeValues, Tree } from "~/models/tree";
 import { TreePicker, TreePickerNode } from "./tree-picker";
 import { treePickerActions, TreePickerAction } from "./tree-picker-actions";
 import { TreeItemStatus } from "~/components/tree/tree";
 
-export const treePickerReducer = (state: TreePicker = createTree(), action: TreePickerAction) =>
+export const treePickerReducer = (state: TreePicker = {}, action: TreePickerAction) =>
     treePickerActions.match(action, {
-        LOAD_TREE_PICKER_NODE: ({ id }) =>
-            setNodeValueWith(setPending)(id)(state),
-        LOAD_TREE_PICKER_NODE_SUCCESS: ({ id, nodes }) => {
-            const [newState] = [state]
+        LOAD_TREE_PICKER_NODE: ({ id, pickerKind }) => {
+            const picker = state[pickerKind] || createTree();
```

Comment:
All four actions do update or create picker with given id. Please, extract common functionality into a function so that we could do
```
LOAD_TREE_PICKER_NODE: ({id, pickerId}) => 
    updateOrCreatePicker(state, pickerId, setNodeValueWith(setPending)(id)),

LOAD_TREE_PICKER_NODE_SUCCESS: ({id nodes, pickerId}) =>
    updateOrCreatePicker(state, pickerId, compose(
        receiveNodes(nodes)(id),
        setNodeValueWith(setLoaded)(id)
    )),

...
```